assets/js/modules/analytics-4/datastore/audiences.js

```diff
@@ -144,42 +114,32 @@ const baseReducer = ( state, { type } ) => {
 };
 
 const baseResolvers = {
-	*getAudiences() {
+	*getAvailableAudiences() {
 		const registry = yield Data.commonActions.getRegistry();
 
-		const audiences = registry.select( MODULES_ANALYTICS_4 ).getAudiences();
+		const audiences = registry
+			.select( MODULES_ANALYTICS_4 )
+			.getAvailableAudiences();
 
-		if ( audiences === undefined ) {
-			yield fetchGetAudiencesStore.actions.fetchGetAudiences();
+		// If available audiences not present, sync the audience in state.
+		if ( audiences === null ) {
+			yield fetchSyncAvailableAudiencesStore.actions.fetchSyncAvailableAudiences();
 		}
-	},
-};
 
-const baseSelectors = {
-	/**
-	 * Gets the property audiences.
-	 *
-	 * @since 1.120.0
-	 *
-	 * @param {Object} state Data store's state.
-	 * @return {(Array|undefined)} An array with audiences objects; `undefined` if not loaded.
-	 */
-	getAudiences( state ) {
-		return state.audiences;
+		return registry.select( MODULES_ANALYTICS_4 ).getAvailableAudiences();
```

Comment:
@techanvil @ankitrox resolvers don't have a return value :) This line can simply be removed.

Reply:
Thanks for spotting that, I totally missed it at review time. I'm including a fix in my current PR.